src/CMakeLists.txt

```diff
@@ -677,8 +677,10 @@ endif()
 if(WIN32)
   # FIXME: This makes no sense...
   #list(APPEND SM_INCLUDE_DIRS ${DIRECTX_INCLUDE_DIR})
-elseif(NOT WITH_SYSTEM_PNG)
-  list(APPEND SM_INCLUDE_DIRS "${SM_EXTERN_DIR}/libpng/include")
+endif()
+
+if(NOT WITH_SYSTEM_PNG)
+  list(APPEND SM_INCLUDE_DIRS "${SM_EXTERN_DIR}/libpng")
```

Comment:
This relates to changes in the includes. Before the specific bundled file was included via `"../extern/libpng/include/png.h"` now we properly include `<png.h>` on all systems. By adding this include path when not compiling against system libpng, we make that possible.